File: Tournonment/BazeEmtahanS1.java
import java.util.Scanner;

public class Main {

    public static void main(String[] args) {
        try (Scanner scanner = new Scanner(System.in)) {
            int s = scanner.nextInt();
            int f = scanner.nextInt();
            int l = scanner.nextInt();
            int x = scanner.nextInt();

            if (x < s) {
                System.out.println("exam did not started!");
            }else if (x >= f) {
                System.out.println("exam finished!");
            }else if (f - x > l){
                System.out.println(l);
            }else {
                System.out.println(f - x);
            }
        }
    }
}

// </Ctrl+Z>

File: Tournonment/BazeEmtahanS1.py
import sys
class Main(object) :
    @staticmethod
    def main( args) :
        try (scanner =  "Python-inputs") :
            s = input()
            f = input()
            l = input()
            x = input()
            if (x < s) :
                print("exam did not started!")
            elif(x >= f) :
                print("exam finished!")
            elif(f - x > l) :
                print(l)
            else :
                print(f - x)

if __name__=="__main__":
    Main.main(sys.argv)
